crates/staged-sync/src/utils/init.rs

```diff
@@ -7,18 +7,52 @@ use reth_db::{
 };
 use reth_primitives::{stage::StageId, Account, Bytecode, ChainSpec, H256, U256};
 use reth_provider::{DatabaseProviderRW, PostState, ShareableDatabase, TransactionError};
-use std::{path::Path, sync::Arc};
-use tracing::debug;
+use std::{fs, io, path::Path, sync::Arc};
+use tracing::{debug, info};
+
+/// The name of the file that contains the version of the database.
+pub const DB_VERSION_FILE_NAME: &str = "version";
+/// The version of the database stored in the [DB_VERSION_FILE_NAME] file in the same directory as
+/// database. Example: `0.1.0-e43455c2`
+pub const DB_VERSION: &str = concat!(env!("CARGO_PKG_VERSION"), "-", env!("VERGEN_GIT_SHA"));
```

Comment:
If we go by commit hash, we would essentially say that every run of the node on a new version requires a resync which is not the case (and terrible for DX), I think we need a more manual approach unfortunately, unless there is some way to generate a hash based on the schema instead.

Reply:
Agree, if this version is indented to detect breaking changes, but we can also start with just keeping track of it.

I'm thinking of putting two versions in this file: the one we've created this DB with, and the current one. First one won't change after the initial start, and the second one will optionally change on every start.

It will help us with debugging user problems related to database. We can do breaking change detection in the future, e.g. by agreeing on bumping major/minor crate version on every breaking change and check it on every `init_db` call.